xwayland/xwm.c

```diff
@@ -583,10 +583,22 @@ static void read_surface_normal_hints(struct wlr_xwm *xwm,
 	memcpy(xsurface->size_hints, &size_hints,
 		sizeof(struct wlr_xwayland_surface_size_hints));
 
-	if ((size_hints.flags & XCB_ICCCM_SIZE_HINT_P_MIN_SIZE) == 0) {
+	bool has_min_size_hints = (size_hints.flags & XCB_ICCCM_SIZE_HINT_P_MIN_SIZE) == 0;
```

Comment:
The condition has been inverted in the process. Should be `!= 0`.

Reply:
i wonder why it even works ...

Reply:
oh because I have inverted the conditions below as well. I should never do PRs while I'm on the verge of falling asleep, sorry.

Reply:
This condition is not inverted, but the `if` condition has been inverted, so it probably doesn't work?

In any case, the variable name is wrong. Could you send a PR please? :)